### PR TITLE
[ADBDEV-3862] Add cancel read endpoint

### PR DESCRIPTION
**Pull request by xardazzzzzz**

### CODE REVIEW

File: external-table/src/libchurl.c [hunk 1]
```diff
@@ -619,6 +619,12 @@ multi_perform(churl_context *context)
 {
 	int			curl_error;
 
+
+	/*
+	 * Allow canceling a query while waiting for input from remote service
+	 */
+	CHECK_FOR_INTERRUPTS();
```
Comment (RekGRpth) [inline external-table/src/libchurl.c]: It was the wrong decision! See reproduction in https://github.com/arenadata/pxf/pull/42

Comment (xardazzzzzz) [reply external-table/src/libchurl.c]: removed

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/rest/PxfReadResource.java [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 import org.greenplum.pxf.api.model.RequestContext;
+import org.greenplum.pxf.service.HttpHeaderDecoder;
 import org.greenplum.pxf.service.RequestParser;
+import org.greenplum.pxf.service.controller.PxfErrorReporter;
 import org.greenplum.pxf.service.controller.ReadService;
+import org.greenplum.pxf.service.spring.PxfThreadPoolTaskExecutor;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
```
Comment (RomaZe) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/rest/PxfReadResource.java]: Please remove imports that we don't use

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/spring/PxfConfiguration.java [hunk 3]
```diff
@@ -16,8 +16,12 @@
 import org.springframework.core.task.TaskExecutor;
 import org.springframework.scheduling.annotation.AsyncAnnotationBeanPostProcessor;
 import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.web.method.support.HandlerMethodReturnValueHandler;
 import org.springframework.web.servlet.config.annotation.AsyncSupportConfigurer;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBodyReturnValueHandler;
+
+import java.util.List;
```
Comment (RomaZe) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/spring/PxfConfiguration.java]: Could you remove please not used imports:  `java.util.List`, `org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBodyReturnValueHandler` and `HandlerMethodReturnValueHandler`

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/rest/PxfReadResource.java (hunk 2 above)
Comment (xardazzzzzz) [reply server/pxf-service/src/main/java/org/greenplum/pxf/service/rest/PxfReadResource.java]: fixed

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/spring/PxfConfiguration.java (hunk 3 above)
Comment (xardazzzzzz) [reply server/pxf-service/src/main/java/org/greenplum/pxf/service/spring/PxfConfiguration.java]: fixed

**Review by RomaZe — APPROVED** [review]

**Pull request merged**